src/flowcean/core/environment/stack.py

```diff
@@ -0,0 +1,36 @@
+from functools import reduce
+from typing import Self
+
+import polars as pl
+
+from .offline import OfflineEnvironment
+
+
+class StackEnvironment(OfflineEnvironment):
+    """Combines multiple OfflineEnvironments into a single one.
+
+    This environment combines multiple OfflineEnvironments into a single one by
+    stacking them vertically. All environments need to share the identical
+    features.
```

Comment:
The fact that they must share identical feature names is only checked in the end, when the data is actually requested. Do we want to do that earlier? Or is this okay for now?

Reply:
This boils down to a general question of error handling: Should we wait and "let the error happen"/polars tell us we did something wrong, or should we check when an environment/transformation is created and throw our own errors? Currently we mostly use the first approach. I'm open to both approaches as long as we keep one through the framework.

Reply:
Let's keep it like this for now...